src/core/pd_repo.js

```diff
@@ -0,0 +1,83 @@
+import mongoose from "mongoose";
+
+import data_chunks from "../constants/pd_constants.js";
+
+import logger from "./pd_logger.js";
+
+class repo {
+  constructor() {
+    this.connectionString = [
+      "mongodb+srv://",
+      data_chunks.DB_user,
+      ":",
+      data_chunks.DB_pass,
+      "@cluster0.br6n1f3.mongodb.net/",
+    ].join("");
+    this.connect();
+  }
+  connect = async () => {
+    await mongoose
+      .connect(this.connectionString)
+      .then(() => {
+        logger.info("CONNECTED");
+      })
+      .catch((err) => {
+        logger.error(err.message);
+      });
+  };
+
+  getData = async (dir) => {
+    return await dir
```

Comment:
read about async, await and promise chaining 
we dont need chaining when we use async await 
ex: .then() .catch()
is not needed assign value to variable and return it 
for catch use try catch

Reply:
applies for all below methods

Reply:
done.